service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala

```diff
@@ -300,23 +322,91 @@ case class RowCountCurator(protected val requestModelValidator: CuratorRequestMo
+                if(model.dimCardinalityEstimate.get.intValue <= FACT_ONLY_LIMIT) {
+                  val count = model.dimCardinalityEstimate.get.intValue
+                  mahaRequestContext.mutableState.put(RowCountCurator.name, count)
+                  mahaRequestLogBuilder.logSuccess()
+                  withResult(parRequestLabel, parallelServiceExecutor, curatorResult)
+                } else {
+                  mahaRequestContext.mutableState.put(RowCountCurator.name, FACT_ONLY_LIMIT)
+                  mahaRequestLogBuilder.logSuccess()
+                  withResult(parRequestLabel, parallelServiceExecutor, curatorResult)
+                }
+              } else {
+                val message = "No row count can be estimated without dim cardinality estimate"
+                mahaRequestLogBuilder.logFailed(message, Option(400))
+                withError(curatorConfig, GeneralError.from(parRequestLabel, message))
+              }
+            } else { // if isFactDriven = true is specified in config, includeRowCount = true
+              val totalRowsCountRequestTry =
+                Try {
+                  //force fact driven
+                  //added row count field
+                  //remove all sorts
+                  sourcePipeline.requestModel.reportingRequest.copy(
+                    selectFields = sourcePipeline.requestModel.reportingRequest.selectFields
+                    , sortBy = IndexedSeq.empty
+                    , includeRowCount = true
+                    , forceDimensionDriven = false
+                    , forceFactDriven = true
+                    , paginationStartIndex = 0
+                    , rowsPerPage = 1
+                    , curatorJsonConfigMap = Map.empty
+                  )
+                }
+              if (totalRowsCountRequestTry.isFailure) {
+                val exception = totalRowsCountRequestTry.failed.get
+                val message = "total rows request failed to generate"
+                mahaRequestLogBuilder.logFailed(s"${message} - ${exception.getMessage}")
+                withError(curatorConfig, GeneralError.from(parRequestLabel, message, exception))
+              } else {
+                val totalRowsRequest = totalRowsCountRequestTry.get
+                val parRequestResult: ParRequestResult = mahaService.executeRequest(mahaRequestContext.registryName
+                  , totalRowsRequest, mahaRequestContext.bucketParams, mahaRequestLogBuilder)
+
+                val populateRowCount: ParRequest[RequestResult] = parRequestResult.prodRun.map(parRequestLabel, ParFunction.fromScala {
+                  requestResult =>
+                    var count = 0
+                    val resRowList = requestResult.queryPipelineResult.rowList
+                    if (!resRowList.isEmpty) {
+                      resRowList.foreach(row => {
+                        count = Option(row.getValue(0)).getOrElse("0").toString.toInt
```

Comment:
rather than doing this use getTotalRowCount method from RowList. In Druid query generator you can use QueryRowList.ROW_COUNT_ALIAS as your field name instead of "Row Count".